wsmaster/codenvy-hosted-api-organization/src/main/java/com/codenvy/organization/api/OrganizationManager.java

```diff
@@ -0,0 +1,200 @@
+
+import com.codenvy.organization.api.event.OrganizationCreatedEvent;
+import com.codenvy.organization.shared.model.Organization;
+import com.codenvy.organization.spi.MemberDao;
+import com.codenvy.organization.spi.OrganizationDao;
+import com.codenvy.organization.spi.impl.OrganizationImpl;
+import com.google.common.collect.Sets;
+
+import org.eclipse.che.api.core.ConflictException;
+import org.eclipse.che.api.core.NotFoundException;
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.core.notification.EventService;
+import org.eclipse.che.commons.lang.NameGenerator;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.util.List;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Facade for Organization related operations.
+ *
+ * @author gazarenkov
+ * @author Sergii Leschenko
+ */
+public class OrganizationManager {
+    private final OrganizationDao organizationDao;
+    private final MemberDao       memberDao;
+    private final EventService    eventService;
+    private final Set<String>     reservedNames;
+
+    @Inject
+    public OrganizationManager(OrganizationDao organizationDao,
+                               MemberDao memberDao,
+                               EventService eventService,
+                               @Named("che.account.reserved_names") String[] reservedNames) {
+        this.organizationDao = organizationDao;
+        this.memberDao = memberDao;
+        this.eventService = eventService;
+        this.reservedNames = Sets.newHashSet(reservedNames);
+    }
+
+    /**
+     * Creates new organization
+     *
+     * @param newOrganization
+     *         organization to create
+     * @return created organization
+     * @throws NullPointerException
+     *         when {@code organization} is null
+     * @throws ConflictException
+     *         when organization with such id/name already exists
+     * @throws ConflictException
+     *         when specified organization name is reserved
+     * @throws ServerException
+     *         when any other error occurs during organization creation
+     */
+    public OrganizationImpl create(Organization newOrganization) throws ConflictException, ServerException {
+        requireNonNull(newOrganization, "Required non-null organization");
+        if (reservedNames.contains(newOrganization.getName().toLowerCase())) {
+            throw new ConflictException(String.format("Organization name '%s' is reserved", newOrganization.getName()));
+        }
+        final OrganizationImpl organization = new OrganizationImpl(NameGenerator.generate("organization", 16),
+                                                                   newOrganization.getName(),
+                                                                   newOrganization.getParent());
+        organizationDao.create(organization);
+        eventService.publish(new OrganizationCreatedEvent(organization));
```

Comment:
I think that publishing this kind of events is logic of manager level but not data. Because it is needed for business logic - not for resolving data integrity issues